### PR TITLE
system-variables: add gogc variables

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@hawkingrei Please involve a technical review and resolve the conflict.

File: system-variables.md [hunk 1]
```diff
@@ -1717,6 +1725,13 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 范围：`[1, 100000]`
 - 这个变量用来控制 General Plan Cache 最多能够缓存的计划数量。
 
+### `tidb_gogc_tuner_threshold`
+
+- 作用域：GLOBAL
+- 是否持久化到集群：否
+- 默认值：`0.6`
+- 范围：`[0, 0.9)`
```
Comment (TomShawn) [inline system-variables.md:1733]:
```suggestion
- 范围：`[0, 18446744073709551615)`
```

Comment (xhebox) [reply system-variables.md:1723]: I think `0.9` is correct. It is the source code wrong. @hawkingrei , correct `MaxValue` of this variable.

**Review by xhebox — APPROVED** [review]

File: system-variables.md [hunk 2]
```diff
@@ -1707,6 +1715,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 开启该变量后，即可在 TiDB Dashboard 中查看查询的图形化执行计划。注意，TiDB Dashboard 只显示变量开启时产生的查询的执行计划。
 - 用 `select tidb_decode_binary_plan('xxx...')` SQL 语句可以从编码后的执行计划解析出具体的执行计划。
 
+### `tidb_gogc_tuner_threshold`
+
+- 作用域：GLOBAL
+- 是否持久化到集群：否
+- 默认值：`0.6`
+- 范围：`[0, 0.9)`
+- 这个变量用来控制 GOGC Tuner 自动调节的最大内存阈值，超过阈值后 GOGC Tuner 会停止工作。
```
Comment (TomShawn) [inline system-variables.md:1724]:
```suggestion
- 这个变量用来控制 GOGC Tuner 自动调节的最大内存用量阈值，超过阈值后 GOGC Tuner 会停止工作。
```

File: system-variables.md [hunk 3]
```diff
@@ -1220,6 +1220,14 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 默认值：`OFF`
 - 这个变量用来控制是否开启 General Plan Cache。
 
+### `tidb_enable_gogc_tuner`
```
Comment (TomShawn) [inline system-variables.md:1223]:
```suggestion
### `tidb_enable_gogc_tuner` <span class="version-mark">从 v6.4.0 版本开始引入</span>
```

File: system-variables.md [hunk 4]
```diff
@@ -1707,6 +1715,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 开启该变量后，即可在 TiDB Dashboard 中查看查询的图形化执行计划。注意，TiDB Dashboard 只显示变量开启时产生的查询的执行计划。
 - 用 `select tidb_decode_binary_plan('xxx...')` SQL 语句可以从编码后的执行计划解析出具体的执行计划。
 
+### `tidb_gogc_tuner_threshold`
```
Comment (TomShawn) [inline system-variables.md:1718]:
```suggestion
### `tidb_gogc_tuner_threshold` <span class="version-mark">从 v6.4.0 版本开始引入</span>
```

File: system-variables.md [hunk 5]
```diff
@@ -1707,6 +1715,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+
+- 作用域：GLOBAL
+- 是否持久化到集群：否
```
Comment (TomShawn) [inline system-variables.md:1721]:
```suggestion
- 是否持久化到集群：否，仅作用于当前连接的 TiDB 实例
```

File: system-variables.md [hunk 6]
```diff
@@ -1220,6 +1220,14 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 默认值：`OFF`
 - 这个变量用来控制是否开启 General Plan Cache。
 
+### `tidb_enable_gogc_tuner`
+
+- 作用域：GLOBAL
+- 是否持久化到集群：否
```
Comment (TomShawn) [inline system-variables.md:1226]:
```suggestion
- 是否持久化到集群：否，仅作用于当前连接的 TiDB 实例
```

File: system-variables.md (hunk 1 above)
Comment (hawkingrei) [reply system-variables.md:1723]: > I think `0.9` is correct. It is the source code wrong. @hawkingrei , correct `MaxValue` of this variable.

OK

File: system-variables.md (hunk 2 above)
Comment (hawkingrei) [reply system-variables.md:1724]: It is not the usage of memory. it is the threshold of memory whether to tune gogc.

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**